### PR TITLE
Fix endianness on floats

### LINKED ISSUE #3050: test/unit/test_float failures on big-endian systems (System Z)
```
>>> MALLOC_PERTURB_=101 /home/travis/build/rizinorg/rizin/build/test/unit/test_float
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
rz_float_new_from_hex_test ERR
[XX] Fail at line 337: new from hex 1: expected true, got false
f32_ieee_format_test ERR
[XX] Fail at line 28: string bit value of 32-bit float: expected 00111111110000000000000000000000, got 00000000000000001100000000111111.
rz_float_detect_spec_test OK
f32_ieee_add_test ERR
[XX] Fail at line 90: test calculating bv value of 1.5f + 0.25f: expected true, got false
f32_ieee_sub_test ERR
[XX] Fail at line 149: test calculating bv value of 1.5f - 0.25f: expected true, got false
f32_ieee_mul_test ERR
[XX] Fail at line 206: Compare Mul of 11.1 * 2.37 == 26.307 ?: expected true, got false
f32_ieee_div_test ERR
[XX] Fail at line 236: Compare Div of 11.1 / 2.37 == 4.6835446 ?: expected true, got false
f32_ieee_fma_test ERR
[XX] Fail at line 379: Fused Mul Add test 4: expected true, got false
rz_float_trunc_test ERR
[XX] Fail at line 288: Truncate Test 1: expected true, got false
rz_float_abs_test ERR
[XX] Fail at line 321: make abs test for positive: expected true, got false
f32_ieee_round_test OK
f32_ieee_sqrt_test ERR
[XX] Fail at line 439: test sqrt 1: expected true, got false
f32_ieee_rem_test ERR
[XX] Fail at line 560: rem test 1: expected true, got false
f32_ieee_special_num_test OK
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
```
See https://app.travis-ci.com/github/rizinorg/rizin/jobs/584188090

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Requires to pass the SYSZ tests.
# SO DO NOT MERGE UNLESS THE UNIT TEST ON SYSZ ARE GOOD ON FLOAT

### CODE REVIEW

**Comment by wargio** [comment]
```
39/113 rizin:unit / float                                 OK              0.14s
```
Nice.

File: librz/util/float/float.c [hunk 1]
```diff
@@ -224,7 +224,11 @@ RZ_API bool rz_float_set_from_single(RZ_NONNULL RzFloat *f, float value) {
 		return false;
 	}
 
+#if RZ_SYS_ENDIAN
```
Comment (ret2libc) [inline librz/util/float/float.c:227]: does this behaviour depend on the build machine or on the binary under analysis?

Comment (XVilka) [reply librz/util/float/float.c:227]: @ret2libc it's used for setting bitvectors from "native" floats, which are machine-dependent, not binary-dependent.

**Review by XVilka — APPROVED** [review]
Thank you!

File: librz/util/float/float.c (hunk 1 above)
Comment (wargio) [reply librz/util/float/float.c:227]: these no. the issue is that on BE machines the stack/heap stored values are in BE format so when we convert them to RzFloat they need to be "read" with the correct endianness.
It's kinda like the endianness for `htol`, depending on the arch you have an empty function or a byte conversion function.

**Pull request merged**